src/librustc/hir/map/hir_id_validator.rs

```diff
@@ -122,9 +122,10 @@ impl<'a, 'hir: 'a> HirIdValidator<'a, 'hir> {
                                        .find(|&(_, &entry)| hir_id == entry)
                                        .expect("no node_to_hir_id entry");
                 let node_id = NodeId::from_usize(node_id);
+                let hir_id = self.hir_map.node_to_hir_id(node_id);
```

Comment:
Hm, is it just me or do we already have the hir_id since we looked up this node_id based on it?

Reply:
We also do have the node_id, but we still recreate it using `from_usize`; I'm not too sure about this spot myself :thinking: .

Reply:
We don't have the `node_id` outside of this `for` loop, it's extracted from the index provided by `enumerate` on `node_to_hir_id`, just before the `NodeId::from_usize` call.
Maybe this predates the map from `HirId`s to `NodeId`s? If so, that should be used instead of an exhaustive search.

As for the `hir_id`, @Mark-Simulacrum is right, `node_to_hir_id` was used to map `hir_id` to `node_id`, so the `let hir_id = ...;` line is redundant.